### PR TITLE
feat(ui): Change stream sidebar to use SelectControl (APP-341)

### PR DESCRIPTION
**Pull request by billyvg**
Also fixes "timesSeen" to be input instead of select dropdown

image

### CODE REVIEW

**Comment by lynnagara** [comment]
The async tag values don't work

**Comment by billyvg** [comment]
@lynnagara https://github.com/getsentry/sentry/pull/8706/commits/b175c25d52b343ad3ebc520dc218458be2b122a0 (added a test too)

**Comment by lynnagara** [comment]
Can you check out how timesSeen is being handled by the search bar with this change? if you start typing `timesSeen:` it gets into a state where it spins forever and issues some invalid queries

File: src/sentry/static/sentry/app/views/stream/tagFilter.jsx [hunk 1]
```diff
@@ -28,103 +32,143 @@ class StreamTagFilter extends React.Component {
...
+      });
     }
   }
 
   componentWillUnmount() {
-    let select = ReactDOM.findDOMNode(this.refs.select);
-    $(select).select2('destroy');
+    if (!this.api) return;
+    this.api.clear();
+    this.api = null;
```
Comment (lynnagara) [inline src/sentry/static/sentry/app/views/stream/tagFilter.jsx]: How come we're manually cleaning this up?

Comment (billyvg) [reply src/sentry/static/sentry/app/views/stream/tagFilter.jsx]: Because it's generally done in ApiMixin, we don't have a HoC for it.

Comment (lynnagara) [reply src/sentry/static/sentry/app/views/stream/tagFilter.jsx]: oh i mean the line `this.api = null`, not the part about clearing outstanding requests

Comment (billyvg) [reply src/sentry/static/sentry/app/views/stream/tagFilter.jsx]: Ah yeah, it's probably un-needed but doesn't hurt to explicitly clear out references

Comment (lynnagara) [reply src/sentry/static/sentry/app/views/stream/tagFilter.jsx]: I don't think we should do it, unless consistently everywhere (which i've not seen)

Comment (billyvg) [reply src/sentry/static/sentry/app/views/stream/tagFilter.jsx]: Why not, what's the harm?

Comment (lynnagara) [reply src/sentry/static/sentry/app/views/stream/tagFilter.jsx]: 1 - it's a line of code that doesn't do anything, you could argue that there's never any harm in those, but there's also no point
2 - in this case it's inconsistent practice with the rest of our codebase

Comment (billyvg) [reply src/sentry/static/sentry/app/views/stream/tagFilter.jsx]: I think it's better to be a bit proactive with cleaning up your references but I've removed it

**Review by lynnagara — APPROVED** [review]

**Pull request merged**